sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/util/DateTimeUtils.scala

```diff
@@ -206,6 +206,22 @@ object DateTimeUtils {
     (day.toInt, micros * 1000L)
   }
 
+  /*
+   * Converts the timestamp to milliseconds since epoc. In spark timestamp values have microseconds
+   * precision, so this conversion is lossy.
+   */
+  def toMillis(us: SQLTimestamp): Long = {
+    val julian_us = us + JULIAN_DAY_OF_EPOCH * MICROS_PER_DAY
```

Comment:
@davies Many thanks !! You are right. We need to write it as milliseconds since epoc.  I will send a update .. Thanks again !!